### ISSUE TITLE
Unable to fetch document

### ISSUE THREAD

**Issue by ashutoshh** [cloudant/python-cloudant#56]
I am saving documents in the cloudant with url as the value of _id field. I am able to save the documents successfully but while retrieving its giving me key error.
Example:
client = Cloudant(USERNAME, PASSWORD,url=cloudantURL)
client.connect()
data={'_id':'https://www.mssdfs.com'}
db = client['dbname']
db.create_document(data)
name='https://www.mssdfs.com'
doc = db[name]
if doc.exists():
    print True
else:
    print False

this gives me key error.

**Comment by alfinkel** [comment]
This is a known bug.  You can find the details and a work around here #32.

**Comment by alfinkel** [comment]
Your simpliest route to take would be to use urllib and replace the line `doc = db[name]` with `doc = Document(db, urllib.quote(name, safe='')`.  Then your check for doc.exists() should work.  

You could also change the line to be `doc = db[urllib.quote(name, safe='')]` and that also should get your doc.exists() check working but it may have an unwanted side effect in that it will add the urlencoded key to the locally cached `db` dictionary.  So you would have 2 keys for the same document, one would be 'https://www.mssdfs.com' and the other would be 'https%3A%2F%2Fwww.mssdfs.com'.

Anyway, this is what I would change your code to for the time being until we get the bug fixed and pushed out in an upcoming release:

```python
import urllib
from cloudant.document import Document
client = Cloudant(USERNAME, PASSWORD,url=cloudantURL)
client.connect()
data={'_id':'https://www.mssdfs.com'}
db = client['dbname']
db.create_document(data)
name='https://www.mssdfs.com'
doc = Document(db, urllib.quote(name, safe=''))
if doc.exists():
    print True
else:
    print False
```

**Comment by alfinkel** [comment]
This bug is resolved and the fix should be made available with the 2.0.0b1 release.
